erpnext/public/js/payment/payments.js

```diff
@@ -27,8 +27,14 @@ erpnext.payments = erpnext.stock.StockController.extend({
 		var me = this;
 	
 		this.dialog.set_primary_action(__("Submit"), function() {
-			me.dialog.hide()
-			me.submit_invoice()
+			//Allow no ZERO payment
+			$.each(me.frm.doc.payments, function (index, data) {
+				if (data.amount != 0) {
+					me.dialog.hide();
+					me.submit_invoice();
+					return;
```

Comment:
This should be configurable, what if user wants to submit an invoice with partial payment?

Reply:
Assuming that he is paying on POS partial payments on this don't think should be !!! But still a partial payment would be different than Zero ... Or can we make it configurable so that is up to user to accept Zero Payments when on POS or not ?